bibliopixel/control/api.py

```diff
@@ -0,0 +1,56 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import urllib
+import getpass, platform, sys, threading
+from .. util import log
+from . control import ExtractedControl
+
+class Api(ExtractedControl):
+    EXTRACTOR = {
+        'keys_by_type': {
+            'direct_command': ['action'],
+            'value_command': ['action', 'value']
+        }
+    }
+
+    def make_value_commands(self, value_command_list):
+        return [{'action': com, 'value': val, 'type': 'value_command'} for (com, val) in value_command_list]
+
+    def make_direct_commands(self, direct_command_list):
+        if direct_command_list == None:
+            return []
+        else:
+            return [{'action': com, 'type': 'direct_command'} for (com) in direct_command_list]
+
+    def make_messages(self, params):
+        parsed = (urllib.parse.parse_qs(params[2:]))
+        vcs = parsed.get('value_command')
+        vals = parsed.get('value')
+        if (vcs != None and vals != None):
```

Comment:
You shouldn't use `== None` and `!= None` - `is None` and `is not None` are a little faster (flake8 will also complain here).

Also - no need for parens, so:

    if not (value_command is None or Value is None):

And there's duplication there - you use the expression `parsed.get('value_command')` twice!

You don't need any of that, nor the condition :-D - check it:

    value_commands = list(zip(parsed.get('value_command', []), parsed.get('value', [])))

Reply:
Also, I'm a bit skeptical about the zip in general.  If one list is a different length from another, it'll silently use the shortest length...

Where do these values come from?  Are they guaranteed to be the same length?

Otherwise, what about: https://docs.python.org/3/library/itertools.html#itertools.zip_longest